lib/machine/fingerprintSupport.ts

```diff
@@ -119,14 +122,14 @@ export const DefaultTargetDiffHandler: FingerprintDiffHandler =
  * @param handler the FingerprintDiffHandler to wrap
  */
 export function diffOnlyHandler(handler: FingerprintDiffHandler): FingerprintDiffHandler {
-    return async (context, diff, aspect) => {
-        if (diff.from && diff.to.sha !== diff.from.sha) {
-            return handler(context, diff, aspect);
-        } else {
-            return {
-                abstain: true,
-            };
-        }
+    return async (context, diffs, aspect) => {
+        const toDiff = _.filter(diffs, diff => diff.from && diff.to.sha !== diff.from.sha);
```

Comment:
Why lodash for a collection.filter?

Reply:
Why not? I'm thinking consistency. Happy to use built in where available if that's preferred?

Reply:
Also, this functional approach gives null safety, which is nice @cdupuis .